client/network/src/peer_info.rs

```diff
@@ -367,6 +396,8 @@ impl NetworkBehaviour for PeerInfoBehaviour {
 			FromSwarm::ExpiredListenAddr(e) => {
 				self.ping.on_swarm_event(FromSwarm::ExpiredListenAddr(e));
 				self.identify.on_swarm_event(FromSwarm::ExpiredListenAddr(e));
+				self.external_addresses
+					.on_swarm_event(FromSwarm::ExpiredListenAddr::<Self::ConnectionHandler>(e));
```

Comment:
Nitpick: if we manually call `self.external_addresses.on_swarm_event` only for two specific events, wouldn't it be nicer to introduce two specialized functions like `register_external_address`/`unregister_external_address` instead of a blank match that never receives other events?